dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,407 @@
+# Minimal OS for Fobnail project
+
+The Fobnail Projects aims to build a USB device capable of verifying
+trustworthiness of the platform it is connected to. The project is described
+more in-depth [here](../docs/index.md).
+
+According to the information presented at TrenchBoot Summit 2021
+[Fobnail: Attestation in Your Pocket](https://youtu.be/xZoCtNV8Qs0?t=5062),
+the minimum OS, together with the connected Fobnail Token, should allow
+determining the state of the platform before the target OS is launched. As we
+can see in the presentation, [heads](https://github.com/osresearch/heads) would
+be responsible for the platform's trust control, while in this document, we want
+to present the results of research on the possibilities of using various
+operating systems.
+
+## OS requirements
+
+We pay attention to the drivers supported by each OS. We need the following
+drivers:
+
+- TPM drivers
+- Drivers required for communicating with Fobnail Token
+  - USB host drivers - at minimum EHCI and xHCI support, however UHCI and OHCI
+    would be good to have
+  - USB EEM driver - emulated Ethernet driver
+  - Network stack with IPv4 support
+
+Also we pay attention to OS security, ability to run in DLME, including
+portability beetween different hardware (a single binary should be able to boot
+on all x86 platforms with support for ACPI) and ability to chainload another,
+target OS.
+
+The table below summarises OS features which are a hard requirement, though they
+could be implemented by us.
+
+| Requirement             | Description                                                        |
```

Comment:
What about ability/ease of obtaining platform metadata?

Reply:
@arturkow2000 I think this is very important requirement, best it would be if we can even test availability of libraries/tools for such purpose. @krystian-hebel thanks, I think this is very good point.

@dpsmith thoughts?

Reply:
@krystian-hebel @pietrushnic Right now obtaining metadata would be problematic on any non-Linux OS because we need to obtain MAC of network adapter. So we virtually need a driver for each possible network card.

Reply:
Are we sure, that MAC is necessary for hardware identification? What will happen for routers with multiple NICs?

Reply:
> Are we sure, that MAC is necessary for hardware identification

IMO we don't need it. EK cert hash together with platform metadata could be used for identification.

> What will happen for routers with multiple NICs?

That wouldn't cause problems, we only take MAC of the first network adapter. [here](https://github.com/fobnail/docs/blob/main/dev-notes/metadata.md#mac-address) we describe how determine which NIC is first.

Reply:
@pietrushnic

IMO we should drop requirement for MAC address. We can't provide drivers for each possible NIC as OS wouldn't be minimal anymore.

I spoke with @krystian-hebel and we could use AIK loaded key name instead of MAC.

Reply:
@arturkow2000 sure, let's not use MAC. I believe I suggested I don't exactly like MAC for hardware identification.

Reply:
@pietrushnic @arturkow2000 So I believe that not using MAC should be added to the TODO list. What about the opening topic of this thread? Shouldn't we add such a requirement?

Reply:
Agree with adding MAC dependency removal to TODO. Other questions should be addressed by @krystian-hebel since he is OP here.

Reply:
I think original question is answered as portability.

Reply:
@krystian-hebel you mean adding requirement of portability to Fobnail Project? Of course, it has to be defined, but could be handled in v2 of this document or in other relevant one.